### ISSUE TITLE
3.2.7 clarifications and a user-agent example

### ISSUE THREAD

**Issue by JamesCatt** [w3c/wcag#1905]
> The working group is interested in feedback about whether there are Components determined by the user agent that should not be in scope.

Would an example of this be the "show password" toggle that appears on password inputs in MS Edge? It only appears when the input is focused. AFAIK there's no way for the author to change this behavior.

> ...information needed to identify that user interface components are available is visible...

Would visible text informing the user of the presence of these components (and how to activate them) be considered sufficient? If so, then perhaps this should be listed as a success technique.

> Hiding the information needed to identify the component is essential.

What would be considered "essential" in this case? It's not currently defined, and I'm having trouble thinking of scenarios that would qualify. The only one I can think of would be something like where space limitations would mean that making the controls visible would overlap other content/controls—but even in this scenario the author could presumably provide an equivalent component somewhere else on the page.

**Comment by alastc** [comment]
Just on this point:
> What would be considered "essential" in this case?

We should add the link to the [definition of essential](https://www.w3.org/TR/WCAG22/#dfn-essential), which is basically: Cannot be achieved in another way.

In this case, perhaps a game that you had to hover-over to flip cards temporarily, in order to select a particular card... I'm not sure, but there was discussion about it and someone else in the group would probably remember why it was added.

**Comment by JamesCatt** [comment]
@alastc now that you mention that, I'm thinking maybe some kind of game where the user has to solve a puzzle by finding/clicking the correct region of an image? Like an interactive _Where's Waldo_ game?

**Comment by alastc** [comment]
> Would an example of this be the "show password" toggle that appears on password inputs in MS Edge? It only appears when the input is focused.

From a quick check, it appears "on input", you have to type something for it to show up, therefore not in scope.

I've also noticed that the main branch has already made "essential" a link to the definition, so that's already done.

**Comment by mbgower** [comment]
Draft official response

Thank you for your response. You provided three questions about possible examples:

1. > Would an example of this be the "show password" toggle that appears on password inputs in MS Edge? It only appears when the input is focused. AFAIK there's no way for the author to change this behavior.

2. > Would visible text informing the user of the presence of these components (and how to activate them) be considered sufficient? 

3. > What would be considered "essential" in this case?

In regard to question 1, as detailed in a prior response, the MS Edge 'show password' mechanism appears on input, not on focus. Therefore, it is not an example of a user agent component that should not be in scope. Further, the mechanism is part of an input component that itself remains visible. So even if the 'show password' mechanism' did appear on focus, the visual indicator that it is available is arguably provided by an input labelled "password". (In other words, someone using Edge would expect the user agent behaviour to consistently occur for every password input.)

In regard to question 2, visible text informing the user of the presence of such components and how to activate them would be considered sufficient to meet the criterion. New content is being added to the Understanding document to clarify this.

in regard to question 3, as previously mentioned, "essential" is now linked to the definition, and new content is being added to the Understanding document to provide a clearer example of essential.

**Comment by alastc** [comment]
Agreed: https://www.w3.org/2022/02/08-ag-minutes.html#t07
